### PR TITLE
BAU: Change error logs to warn

### PR DESCRIPTION
**Pull request by CarlyG55**
## What

These logs are all a result of bad requests, so we should be logging them as warns rather than errors

## How to review

Check that all updated logs are in fact a result of a bad request, and not an error our side

## Checklist

- [ ] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

**Review by isaac-GDS — DISMISSED** [review]

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/QueryParamsAuthorizeValidator.java [hunk 1]
```diff
@@ -47,7 +47,9 @@ public Optional<AuthRequestError> validate(AuthenticationRequest authRequest) {
         ClientRegistry client = getClientFromDynamo(clientId);
 
         if (!client.getRedirectUrls().contains(authRequest.getRedirectionURI().toString())) {
-            LOG.warn("Invalid Redirect URI in request {}", authRequest.getRedirectionURI());
+            logErrorInProductionOrWarnOtherwise(
+                    String.format(
+                            "Invalid Redirect URI in request %s", authRequest.getRedirectionURI()));
```
Comment (CarlyG55) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/QueryParamsAuthorizeValidator.java:52]: I imagine we want the same behaviour on this as others, even though it was warn initially?

Comment (billrdunn) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/QueryParamsAuthorizeValidator.java:52]: Could use `logErrorInProductionOrWarnOtherwise()` in `generateBadRequestResponse()`?

Comment (CarlyG55) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/QueryParamsAuthorizeValidator.java:52]: Hmm, we could, but we'd likely lose a bit of information on the way (or have to do a bunch of refactoring...)

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/BaseAuthorizeValidator.java [hunk 2]
```diff
@@ -44,4 +44,12 @@ ClientRegistry getClientFromDynamo(String clientId) {
         }
         return client;
     }
+
+    protected void logErrorInProductionOrWarnOtherwise(String logMessage) {
```
[the remark this answers is not in the document]
Comment (CarlyG55) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/BaseAuthorizeValidator.java]: have made this change but done it manually to update all the references (I'm bad with words 😆)

**Review by billrdunn — APPROVED** [review]

**Comment by andrew-moores** [comment]
@CarlyG55 @billrdunn Can this PR be merged now or is it no longer needed?

**Comment by CarlyG55** [comment]
@andrew-moores Apologies, this is now merged!

**Pull request merged**